src/mercury_engine_data_structures/formats/__init__.py

```diff
@@ -1,5 +1,8 @@
 from typing import Type
 
+from mercury_engine_data_structures.formats.bapd import Bapd
+
+# this must come before any formats that use BaseResource!
```

Comment:
Whats the reasoning for this? It sounds like something that can get up messed really easily with any formatter
(Also capitalize `This` please)

Reply:
... precommit fucked it up, check the  original commit. this is why tests fail btw.

Reply:
it needs to be out of order or else we get a circular import on BaseResource